optimum/onnxruntime/modeling_decoder.py

```diff
@@ -336,8 +336,7 @@ def prepare_past_key_values(
             dtype = constructor.float16 if self.use_fp16 else constructor.float32
 
             # TODO: find a way to better handle this controlflow, this is EXTREMELY UGLY.
-            # "1" is the dummy sequence length
-            if self.model_type == "bloom":
+            if self.model_type == "bloom" and not check_if_transformers_greater("4.44"):
```

Comment:
in modeling, idealy we should use the transformers version used for export, but idk if there's anyway to access that from an onnx model.

Reply:
what if a model was previously exported with a lower version ?

Reply:
it will fail, we need something similar to https://github.com/huggingface/optimum-intel/pull/877#discussion_r1732369193

Reply:
Could this be added (can be done in a following PR) to make sure we don't lose compatibility with previously exported model ? Can be done like optimum-intel, an other option could be to check the graph to check what input shape is expected

Reply:
using input shapes makes sense, I'm testing it rn

Reply:
done!